exchanges/bitmex/bitmex_websocket.go

```diff
@@ -141,338 +152,321 @@ func (b *Bitmex) wsReadData() {
 }
 
 func (b *Bitmex) wsHandleData(respRaw []byte) error {
-	quickCapture := make(map[string]interface{})
-	err := json.Unmarshal(respRaw, &quickCapture)
+	// We don't need to know about errors, since we're using looking optimistically into the json
+	msg, _, _, err := jsonparser.Get(respRaw, "[3]")
 	if err != nil {
-		return err
+		return fmt.Errorf("unknown message format: %s", respRaw)
+	}
+	op, _ := jsonparser.GetString(msg, "request", "op")
+	errMsg, _ := jsonparser.GetString(msg, "error")
+	success, _ := jsonparser.GetBoolean(msg, "success")
+	version, _ := jsonparser.GetString(msg, "version")
+	switch {
+	case version != "":
+		op = "open"
+		fallthrough
```

Comment:
This fallthrough tripped me up because `errMsg` is `""` and `success` is `false` when this happens. That is on me, I just wanted to tell a cool story